### DISCUSSION TITLE
Option. Gameplay field scaling with no render area restrictions.

### DISCUSSION THREAD

**Discussion by okolenmi** [ppy/osu#20673] [Ideas] [answered]
With new mods like Deflate, Repel and especially Barrel Roll (this mod sometimes can take the circle out of the playing field) it's very uncomfortable to play with limited playfield. Circles on edges is much harder to read as they have only partial approach circle or partial circle in case of deflate, or NO CIRCLE with Barrel Roll + Hidden + Repel (it's very rare case but still stupid).
Playfield without render area limit is a nice way to fix it.
It's very easy to implement. Something like this (not sure it's legit code for C#):
```
//if you don't want to deform objects
float scaling = Math.Min(Scaling.X, Scaling.Y);
foreach ( HitObject ho in HitObjects)
{
    ho.Position.X *= scaling; 
    ho.Position.Y *= scaling;
}
HitObjects.Where(ho => ho is Slider).ForEach(s => s.SliderPoints.ForEach(sp => {sp.Position.X *= scaling; sp.Position.Y *= scaling;}););
General.CircleSize /= scaling; //or another way. Not sure that it's proportional and can be divided directly
```
For replays it's possible to create new "scaling" variable - very interesting solution where player can play replay with the same scale as a player of map. Or scaled coordinates (1/scaling).

I just **want to remind** that it's possible now to disable render restriction for one axis by using scaling only for one side.

This post is not about problems of mods. The main point is playfield restriction.

It's all.

**Comment by peppy** [comment, marked as answer]
https://github.com/ppy/osu/discussions/19985

Please search first.

**Reply by okolenmi** [reply]
Searched but did not find. #19985 has no correct answer. You still can abuse only one axis scaling for extra vision in other, so it's not "intensional", I think.

**Reply by peppy** [reply]
Feel free to discuss there.
